### PR TITLE
Truncate device name to 15 characters

### LINKED ISSUE #503: Config filenames can sometimes be too long
Network interface names have a max length of 15. Truncate the config filenames to this length so they can be seamlessly imported with `wg`.

### PR DESCRIPTION
**Pull request by princemaple**
close #503

### CODE REVIEW

**Comment by princemaple** [comment]
Now that the cache is working for unit tests (probably got issues with the double run) it's getting even better

image

**Comment by princemaple** [comment]
EVEN BETTER 
image

File: apps/fz_http/lib/fz_http/devices.ex [hunk 1]
```diff
@@ -134,8 +134,7 @@ defmodule FzHttp.Devices do
       use_site_mtu
       use_site_persistent_keepalive
     )a
-    |> Enum.map(fn field -> {field, Device.field(changeset, field)} end)
-    |> Map.new()
+    |> Map.new(&{&1, get_field(changeset, &1)})
```
Comment (jamilbk) [inline apps/fz_http/lib/fz_http/devices.ex:137]: 👍

File: docs/docs/user-guides/add-users.md [hunk 2]
```diff
@@ -14,9 +14,10 @@ them access to your network. This is done through the Web UI.
 ## Web UI
 
 Add a user by clicking the "Add User" button under `/users`. You will be asked
-to specify an email and a password for the user. Firezone can also integrate and sync
-with an identity provider to automatically grant access to users in your
-organization. See [Authenticate]({%link docs/authenticate/index.md%}) for more information.
+to specify an email and a password for the user. Firezone can also integrate and
```
Comment (jamilbk) [inline docs/docs/user-guides/add-users.md:17]: Thanks for the fix!

**Review by jamilbk — APPROVED** [review]

**Comment by jamilbk** [comment]
> Now that the cache is working for unit tests (probably got issues with the double run) it's getting even better

Hah. It'll get even better when some of the [concurrency limitations I introduced](https://github.com/firezone/firezone/blob/master/apps/fz_http/lib/fz_http/devices.ex#L27) get fixed so that we can revert the tests involving devices back to `async: true`. I had to set them to sequential mode to prevent deadlock issues :-/. Not my proudest moment as an engineer 😅

**Pull request merged**